internal/handler_factory.go

```diff
@@ -6,32 +6,54 @@ import (
 	"github.com/wavefronthq/wavefront-sdk-go/internal/sdkmetrics"
 )
 
-type HandlerFactory struct {
+type SenderFactory struct {
 	metricsReporter    Reporter
 	tracesReporter     Reporter
 	flushInterval      time.Duration
 	bufferSize         int
-	lineHandlerOptions []LineHandlerOption
+	lineHandlerOptions []BatchAccumulatorOption
+	registry           sdkmetrics.Registry
 }
 
-func NewHandlerFactory(
+func NewSenderFactory(
 	metricsReporter,
 	tracesReporter Reporter,
 	flushInterval time.Duration,
 	bufferSize int,
-	registry sdkmetrics.Registry) *HandlerFactory {
-	return &HandlerFactory{
+	registry sdkmetrics.Registry) *SenderFactory {
+	return &SenderFactory{
+		registry:        registry,
 		metricsReporter: metricsReporter,
 		tracesReporter:  tracesReporter,
 		flushInterval:   flushInterval,
 		bufferSize:      bufferSize,
-		lineHandlerOptions: []LineHandlerOption{
+		lineHandlerOptions: []BatchAccumulatorOption{
 			SetRegistry(registry),
 		},
 	}
 }
 
-func (f *HandlerFactory) NewPointHandler(batchSize int) *RealLineHandler {
```

Comment:
nit: could the batchSize be added to the sender factory? Are there plans to make it configurable by sender?